posts/navarroaxel/11.npm-query-for-installed-packages.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Querying installed packages with npm query
+published: true
+description: Check your installed dependencies using CSS selectors format.
+tags: 'javascript, webdev, node, react'
+cover_image: ./assets/npm-query-cover.jpg
+id: 1209907
+---
+
+Do you know if you use npm dependencies with `postinstall` scripts? Do you have a package installed twice? Since version 8.18.0 we can use [`npm query`](https://docs.npmjs.com/cli/v8/commands/npm-query) to find packages in the `node_modules` directory matching a specific query.
+This is helpful in identifying possible issues and fixing them.
+
+## Why is a postinstall script important?
+
+The `postinstall` script allows a package to run code when it's installed and this could be used in a malicious way. The npm security team receives [reports of malware](https://docs.npmjs.com/reporting-malware-in-an-npm-package) in the registry and works to keep the ecosystem safe, but did you check which dependencies are running in `postinstall` scripts? Do you trust them?
+
+To uninstall all dependencies with a `postinstall` script use the following command:
```

Comment:
Shouldn't you show an example that lists all these dependencies and not just a command that uninstalls things? Or at least include some warning message. Somebody might copy and execute without noticing it's an uninstall script.